### PR TITLE
fest: Environment Resource - configuration variables read & delete

### PR DESCRIPTION
**Pull request by GiliFaroEnv0**
### Issue & Steps to Reproduce / Feature Request
Environment Resource - configuration variables read & delete

### Solution

Added Read logic for configuration variables 
Added delete and update logic for configuration variables

### CODE REVIEW

**Comment by GiliFaroEnv0** [comment]
[Environment Resource - configuration variables read & delete](https://trello.com/c/3d8l8f7B/4839-environment-resource-configuration-variables-read-delete)

**Review by eranelbaz — COMMENTED** [review]
Looks good
please also add this use case to the integration tests
and generate the docs

Also your CI is failing

File: env0/data_environment.go [hunk 1]
```diff
@@ -91,9 +91,8 @@ func dataEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta inter
 			return err
 		}
 	}
-
 	d.SetId(environment.Id)
-	setEnvironmentSchema(d, environment)
+	setEnvironmentSchema(d, environment, client.ConfigurationChanges{})
```
[the remark this answers is not in the document]
Comment (GiliFaroEnv0) [reply env0/data_environment.go:95]: hate this 
also no Ternary Operator ?:

as for the integration tests will be added on a different ticket to unblock the QA PARTY 🎉

[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/data_environment.go:95]: Thats not a blocker, as those are terraform files we need to update

[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/data_environment.go:95]: update this file 
https://github.com/env0/terraform-provider-env0/blob/main/tests/integration/012_environment/main.tf
add the configuration part for the `resource` to create it and also `data` to read it
then expose a `output` and you need to assert the `output` is as you expect using this file
https://github.com/env0/terraform-provider-env0/blob/main/tests/integration/012_environment/expected_outputs.json

really small part

File: env0/resource_environment.go [hunk 2]
```diff
@@ -183,22 +190,45 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
 	if environment.AutoDeployOnPathChangesOnly != nil {
 		d.Set("auto_deploy_on_path_changes_only", *environment.AutoDeployOnPathChangesOnly)
 	}
-	//TODO: env\terraform variables
+	setEnvironmentSchemaConfiguration(d, configurationVariables)
+}
+
+func setEnvironmentSchemaConfiguration(d *schema.ResourceData, configurationVariables []client.ConfigurationVariable) {
```
[the remark this answers is not in the document]
Comment (GiliFaroEnv0) [reply env0/resource_environment.go]: https://github.com/env0/terraform-provider-env0/pull/182/commits/4dc342f20e911ac820b84fd463a224cb8d2715c0

File: env0/data_environment.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (GiliFaroEnv0) [reply env0/data_environment.go:95]: https://trello.com/c/48vpwwFW/4930-tf-provider-add-integration-tests-to-environments-configuration-variables

[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/data_environment.go:95]: Why not here?
thats really small change

[the remark this answers is not in the document]
Comment (GiliFaroEnv0) [reply env0/data_environment.go:95]: @eranelbaz its not tried alot but its not

[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/data_environment.go:95]: lets work this out in a next pr

**Review by eranelbaz — APPROVED** [review]

**Pull request merged**